pkg/client/secrets.go

```diff
@@ -48,6 +48,75 @@ func (c *HTTPClient) GetSecret(ctx context.Context, storeName, id, version strin
 	return secret, nil
 }
 
+func (c *HTTPClient) DeleteSecret(ctx context.Context, storeName, id string) error {
+	reqURL := fmt.Sprintf("%s/%s/%s", withURLStore(c.config.URL, storeName), secretsPath, id)
+
+	response, err := deleteRequest(ctx, c.client, reqURL)
+	if err != nil {
+		return err
+	}
+
+	defer closeResponse(response)
+	err = parseResponse(response, new(string))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *HTTPClient) RestoreSecret(ctx context.Context, storeName, id string) error {
+	reqURL := fmt.Sprintf("%s/%s/%s/restore", withURLStore(c.config.URL, storeName), secretsPath, id)
+
+	response, err := putRequest(ctx, c.client, reqURL, nil)
+	if err != nil {
+		return err
+	}
+
+	defer closeResponse(response)
+	err = parseResponse(response, new(string))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *HTTPClient) DestroySecret(ctx context.Context, storeName, id string) error {
+	reqURL := fmt.Sprintf("%s/%s/%s/destroy", withURLStore(c.config.URL, storeName), secretsPath, id)
+
+	response, err := putRequest(ctx, c.client, reqURL, nil)
+	if err != nil {
+		return err
+	}
+
+	defer closeResponse(response)
+	err = parseResponse(response, new(string))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *HTTPClient) GetDeletedSecret(ctx context.Context, storeName, id string) (*types.SecretResponse, error) {
+	secret := &types.SecretResponse{}
+	reqURL := fmt.Sprintf("%s/%s/%s/deleted", withURLStore(c.config.URL, storeName), secretsPath, id)
```

Comment:
`GetDeleted` and `ListDeleted` are just Get and List but with a query param. This is the example for Eth1 in the handler:
```
getDeleted := request.URL.Query().Get("deleted")
	var eth1Acc *entities.ETH1Account
	if getDeleted == "" {
		eth1Acc, err = eth1Store.Get(ctx, getAddress(request))
	} else {
		eth1Acc, err = eth1Store.GetDeleted(ctx, getAddress(request))
	}
```

Reply:
Applying same format